### PR TITLE
[3.2] Fix issues reported by static analyse

### PR DESCRIPTION
**Pull request by romainneutron**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

Follow-up #21801

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/ApcuAdapter.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function __construct($namespace = '', $defaultLifetime = 0, $version = nu
             CacheItem::validateKey($version);
 
             if (!apcu_exists($version.'@'.$namespace)) {
-                $this->clear($namespace);
```
Comment (romainneutron) [inline src/Symfony/Component/Cache/Adapter/ApcuAdapter.php]: @nicolas-grekas The `clear` method has no argument; is it a bug?

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 2]
```diff
@@ -374,13 +374,7 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $lowercase = true)
                         $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE;
                     }
 
-                    if ($strict = $arg->getAttribute('strict')) {
-                        $strict = XmlUtils::phpize($strict);
-                    } else {
-                        $strict = true;
-                    }
-
-                    $arguments[$key] = new Reference($arg->getAttribute('id'), $invalidBehavior, $strict);
```
Comment (romainneutron) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: `Reference` has lost its third argument

File: src/Symfony/Component/Cache/Adapter/ApcuAdapter.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Component/Cache/Adapter/ApcuAdapter.php]: maybe it was expected to call ``doClear`` directly ?

Comment (romainneutron) [inline src/Symfony/Component/Cache/Adapter/ApcuAdapter.php]: Updated

**Comment by nicolas-grekas** [comment]
Thank you @romainneutron.

**Comment by acrobat** [comment]
@romainneutron which tool is used to do this analysis?

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php [hunk 3]
```diff
@@ -296,7 +296,7 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
             if (__CLASS__ !== get_class($this)) {
                 $r = new \ReflectionMethod($this, __FUNCTION__);
                 if (__CLASS__ !== $r->getDeclaringClass()->getName()) {
-                    @trigger_error(sprintf('Method %s() will have a 6th `$format = null` argument in version 4.0. Not defining it is deprecated since 3.2.', get_class($this), __FUNCTION__), E_USER_DEPRECATED);
+                    @trigger_error(sprintf('Method %s::%s() will have a 6th `$format = null` argument in version 4.0. Not defining it is deprecated since 3.2.', get_class($this), __FUNCTION__), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: @HeahDude organic analyses failure: `get_class($this) !== __CLASS__` when $this is a child class of AbstractNormalizer (which it always is in this case ;) )

**Comment by romainneutron** [comment]
@acrobat PHPStan https://github.com/phpstan/phpstan

**Pull request merged**